### DISCUSSION TITLE
Pandas style to PDF using latex

### DISCUSSION THREAD

**Discussion by a0th** [quarto-dev/quarto-cli#5674] [Q&A]
Im trying to display a HTML object generated by pandas.style.background_gradient.

This code:
```python
import pandas as pd

data = {'A': [1,2,3,4],
        'B': [1,2,3,4],
        'C': [3,2,3,4],
        'D': [4,2,3,4]}

df = pd.DataFrame(data)

styled_df = df.style.background_gradient()
styled_df
```

Produces this output in VSCode. Using jupyter will yield a similar result.
![image](https://github.com/quarto-dev/quarto-cli/assets/1168223/1f7d7831-1179-4a9a-986d-ada676aa28fe)

When producing a PDF with latex, the style is ignored:

![image](https://github.com/quarto-dev/quarto-cli/assets/1168223/586d3010-11dc-432a-be0f-b22b71548383)

When I use `styled_df.to_latex()`, I get:

![image](https://github.com/quarto-dev/quarto-cli/assets/1168223/8766a35e-50f1-404c-9d58-9676e0975691)

If I try to print it instead with `print(styled_df.to_latex())`, I get:

![image](https://github.com/quarto-dev/quarto-cli/assets/1168223/ab6aadd8-9ee9-4555-b8ce-16d1beb1e45e)

Finally, if I use `#| output: asis`, I run into a compilation error:
```
Undefined control sequence.
l.168 0 & \background
                     -color#fff7fb \color#000000 1 & \background-color#fff7f... 
```

If I try using .bar, I fall into another problem.

This code:
```python
#| output: asis
import pandas as pd

data = {'A': [1,2,3,4],
        'B': [1,2,3,4],
        'C': [3,2,3,4],
        'D': [4,2,3,4]}

df = pd.DataFrame(data)

styled_df = df.style.bar()
print(styled_df.to_latex())
```
Gives this compilation error:
```
Undefined control sequence.
l.168 0 & \width
                10em \backgroundlinear-gradient(90deg, #d65f5f 25.0%, transp... 
```

Is there a way to print this table with the style options to PDF?

**Comment by cscheid** [comment]
> Is there a way to print this table with the style options to PDF?

In general, these custom library calls will require you to read the documentation of said libraries. In this case, Pandas is emitting commands that require specific packages. Look at https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.to_latex.html under "Latex Packages", 

and then add the required packages using something like

````
format:
  pdf:
    include-in-header: 
      text: |
        \usepackage{whatever-package-is-required-1}
        \usepackage{whatever-package-is-required-2}
````

It's possible that this won't work in general, since not all LaTeX packages are compatible with one another. But that's the route you'll need to take.

**Comment by olofahlen** [comment]
I ran into this today as well. It looks like a bug to me. I created this issue
https://github.com/pandas-dev/pandas/issues/54082
